Fix confusing OrcCorruptionException thrown by ORC metadata reader

sdruzkin opened the pull request:
## Description
Make the ORC ExceptionWrappingMetadataReader throw OrcCorruptionException only
in case of protobuf exceptions, and make it re-throw the rest of exceptions.

Differential Revision: D72724589

## Motivation and Context
Old implementation re-throws any downstream IO exceptions, like throttling or transport error, as
OrcCorruptionException, which is super confusing. This change makes the metadata reader much
more selective.

## Impact
No impact.

## Test Plan
Added new unit tests.

## Contributor checklist


- [x] Please make sure your submission complies with our [contributing guide](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md), in particular [code style](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style) and [commit standards](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#commit-standards).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

sdruzkin commented:
@NikhilCollooru can you please take a look at this PR?

NikhilCollooru reviewed (commented):
Unfortunately we don't use mockito in presto code. Please use interface and have test class implement that and use it. We have many examples of that.

sdruzkin commented:
> Unfortunately we don't use mockito in presto code. Please use interface and have test class implement that and use it. We have many examples of that.

Updated PR: replaced mockito with a custom mock class.

NikhilCollooru commented on presto-orc/src/test/java/com/facebook/presto/orc/metadata/TestExceptionWrappingMetadataReader.java:
```diff
@@ -0,0 +1,193 @@
+        expectThrows(OrcCorruptionException.class, lambda);
+    }
+
+    private static class MockMetadataReader
```
nit: Please use `TestMetadataReader`

NikhilCollooru commented on presto-orc/src/test/java/com/facebook/presto/orc/metadata/TestExceptionWrappingMetadataReader.java:
```diff
@@ -0,0 +1,193 @@
+                new RuntimeException("test runtime exception"),
+                new IOException("test io exception"),
+                new InvalidProtocolBufferException("test protobuf exception"));
+        private int currentExceptionIdx;
```
nit: `currentExceptionIndex`

NikhilCollooru reviewed (approved):
few nits. but lgtm

sdruzkin commented:
Unrelated test keep failing: 
```
TestResourceManagerClusterStatusSender.testQueryHeartbeat:140 expected [true] but found [false]
```

sdruzkin commented:
Applied suggested changes.
